armi/physics/neutronics/globalFlux/tests/test_globalFluxInterface.py

```diff
@@ -0,0 +1,150 @@
+"""
+Tests for generic global flux interface.
+"""
+import unittest
+
+import numpy
+
+from armi import settings
+from armi.reactor import reactors
+
+from armi.physics.neutronics.globalFlux import globalFluxInterface as gi
+from armi.reactor.tests import test_reactors
+from armi.tests import ISOAA_PATH
+from armi.nuclearDataIO import isotxs
+
+
+class MockParams:
+    pass
+
+
+class MockCore:
+    def __init__(self):
+        self.geomType = "spiral"
+        self.symmetry = "full"
+        self.p = MockParams()
+
+
+class MockReactor:
+    def __init__(self):
+        self.core = MockCore()
+        self.o = None
+
+
+class MockGlobalFluxInterface(gi.GlobalFluxInterface):
+    """
+    Add fake keff calc to a the general gf interface.
+    
+    This simulates a 1000 pcm keff increase over 1 step.
+    """
+
+    def interactBOC(self, cycle=None):
+        gi.GlobalFluxInterface.interactBOC(self, cycle=cycle)
+        self.r.core.p.keff = 1.00
+
+    def interactEveryNode(self, cycle, node):
+        gi.GlobalFluxInterface.interactEveryNode(self, cycle, node)
+        self.r.core.p.keff = 1.01
+
+
+class MockGlobalFluxWithExecuters(gi.GlobalFluxInterfaceUsingExecuters):
+    def getExecuterCls(self):
+        return MockGlobalFluxExecuter
+
+
+class MockGlobalFluxExecuter(gi.GlobalFluxExecuter):
+    def _readOutput(self):
+        class MockOutputReader:
```

Comment:
It seems like there are 2 ways to change self.r.core.p.keff , one here, and one in interact every node. Can you add a comment explaining how they relate.

Reply:
Added docstring along those lines.